### PR TITLE
feat: support project attributes via CLI

### PR DESCRIPTION
**Pull request by jaysh**
- [X] Follows [CONTRIBUTING](https://github.com/snyk/snyk/blob/master/.github/CONTRIBUTING.md) rules

#### What does this PR do?

(First CLI PR! Welcome all feedback.)

This PR adds 3 new flags for setting project attributes from the CLI:

* business criticality (set via `--business-criticality`)
* environment (set via `--environment`)
* lifecycle (set via `--lifecycle`)

Each one can only have a value from the documentation (this is validated locally). They map to these fields in the UI:

image

The values that can be used are the same set that the UI allows you to select from. No free-form values are allowed for any of these 3 fields.

#### Where should the reviewer start?

Start with the docs and tests, this should explain the expected behaviour.

#### How should this be manually tested?

Use `snyk monitor` and pass each of the flags as above. If a flag is not provided, it is not updated on the project. If it is provided, the entire list provided (including an empty list) will overwrite what is already set - no merging occurs.

#### What are the relevant tickets?

`RHINO-237`

#### Screenshots

Inlined above.

### CODE REVIEW

File: test/jest/acceptance/cli-args.spec.ts [hunk 1]
```diff
@@ -278,6 +278,140 @@ describe('cli args', () => {
...
+      `monitor --business-criticality=high,invalid`,
+      {
+        env,
+      },
+    );
+    expect(stdout).toMatch(
+      '1 invalid business-criticality: invalid. Possible values are: critical, high, medium, low',
+    );
+    expect(code).toEqual(2);
+  });
```
[the remark this answers is not in the document]
Comment (jaysh) [reply test/jest/acceptance/cli-args.spec.ts:305]: Good to know, thanks. I will find a minimal set as a smoke test (e.g. to ensure we are actually calling `generateTags`) and migrate the rest to unit tests.

[the remark this answers is not in the document]
Comment (jaysh) [reply test/jest/acceptance/cli-args.spec.ts]: Done!

**Review by ghost — APPROVED** [review]

**Review by ghost — APPROVED** [review]
Approved. Do these attributes affect project uniqueness in anyway?

**Comment by jaysh** [comment]
> Approved. Do these attributes affect project uniqueness in anyway?

Nope. They're metadata-only.

**Pull request merged**